app/client/cli/peer/list.go

```diff
@@ -0,0 +1,106 @@
+package peer
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/pokt-network/pocket/app/client/cli/helpers"
+	"github.com/pokt-network/pocket/logger"
+	"github.com/pokt-network/pocket/p2p/debug"
+	"github.com/pokt-network/pocket/shared/messaging"
+)
+
+var (
+	listCmd = &cobra.Command{
+		Use:   "list",
+		Short: "Print addresses and service URLs of known peers",
+		RunE:  listRunE,
+	}
+
+	ErrRouterType = fmt.Errorf("must specify one of --staked, --unstaked, or --all")
+)
+
+func init() {
+	PeerCmd.AddCommand(listCmd)
+}
+
+func listRunE(cmd *cobra.Command, _ []string) error {
+	var routerType debug.RouterType
+
+	bus, err := helpers.GetBusFromCmd(cmd)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case stakedFlag:
+		if unstakedFlag || allFlag {
+			return ErrRouterType
+		}
+		routerType = debug.StakedRouterType
+	case unstakedFlag:
+		if stakedFlag || allFlag {
+			return ErrRouterType
+		}
+		routerType = debug.UnstakedRouterType
+	// even if `allFlag` is false, we still want to print all peers
+	default:
+		if stakedFlag || unstakedFlag {
+			return ErrRouterType
+		}
+		routerType = debug.AllRouterTypes
+	}
+
+	debugMsg := &messaging.DebugMessage{
+		Action: messaging.DebugMessageAction_DEBUG_P2P_PEER_LIST,
+		Type:   messaging.DebugMessageRoutingType_DEBUG_MESSAGE_TYPE_BROADCAST,
+		Message: &anypb.Any{
+			Value: []byte(routerType),
+		},
+	}
+	debugMsgAny, err := anypb.New(debugMsg)
+	if err != nil {
+		return fmt.Errorf("creating anypb from debug message: %w", err)
+	}
+
+	if localFlag {
```

Comment:
See my comments in #801. They're similar to this as well

Reply:
@Olshansk can you clarify which comment you are referring to? And what you think should change here?

Reply:
Doing a `ctrl f` for `go build` is a good starting point here.

<img width="575" alt="Screenshot 2023-07-25 at 3 47 51 PM" src="https://github.com/pokt-network/pocket/assets/1892194/0c96a69f-44fc-4164-b641-4018d06e2725">

Reply:
@Olshansk the reason I am tackling this first is that this is the base of #801 as such I feel it would be difficult to address #801 when the changes may need to be downwards propagated. Will go through #801 alongside from now on

Reply:
Thank you. This one is on me

Reply:
Have gone through #801 and I think I have already addressed most of the relavent comments in this PR as is